File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        
        int chars[] = new int[26];
        
        for(char c: s.toCharArray()){
            chars[c-'a']++;
        }
        
        for(int i=0; i<s.length(); i++){
            char c = s.charAt(i);
            if(chars[c-'a'] == 1) return i;
        }
        
        
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        
        vector<int> chars(26); 
        for(char c: s) chars[c-'a']++;
        
        for(int i=0; i<s.size(); i++){
            char c = s[i];
            if(chars[c-'a'] == 1) return i;
        }
        
        
        return -1;
    }
};